components/textfield/index.css

```diff
@@ -10,116 +10,445 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+.spectrum-Textfield-input-wrapper {
+  grid-row: 2;
+  grid-column: 1 / span 3;
+  box-sizing: border-box;
+  position: relative;
+
+  /*** Focus Indicator - Unfocused Base Styles ***/
+  &::after {
+    block-size: calc(100% + (var(--mod-textfield-focus-indicator-gap, var(--spectrum-textfield-focus-indicator-gap)) * 2));
+    inline-size: calc(100% + (var(--mod-textfield-focus-indicator-gap, var(--spectrum-textfield-focus-indicator-gap)) * 2));
     position: absolute;
     left: 0;
     right: 0;
-    bottom: 0;
     top: 0;
-    margin: calc(var(--spectrum-textfield-m-texticon-focus-ring-gap) * -1);
-    border-radius: calc(var(--spectrum-textfield-texticon-border-radius) + var(--spectrum-textfield-m-texticon-focus-ring-gap));
+    margin-block-start: calc((var(--mod-textfield-focus-indicator-gap, var(--spectrum-textfield-focus-indicator-gap))
+                              + var(--mod-textfield-focus-indicator-width, var(--spectrum-textfield-focus-indicator-width)))
+                              * -1);
+    margin-inline-start: calc((var(--mod-textfield-focus-indicator-gap, var(--spectrum-textfield-focus-indicator-gap))
+                              + var(--mod-textfield-focus-indicator-width, var(--spectrum-textfield-focus-indicator-width)))
+                              * -1);
+    pointer-events: none;
+    content: '';
     border-color: transparent;
-    transition: box-shadow var(--spectrum-global-animation-duration-100) ease-in-out, border-color var(--spectrum-global-animation-duration-100) ease-in-out;
+    border-style: solid;
+    border-width: 0;
+    border-radius: calc(var(--mod-textfield-corner-radius, var(--spectrum-textfield-corner-radius))
+                        + var(--mod-textfield-border-width, var(--spectrum-textfield-border-width))
+                        + var(--mod-textfield-focus-indicator-gap, var(--spectrum-textfield-focus-indicator-gap)));
   }
 
-  &.spectrum-Textfield--quiet {
-    &:after {
+  .is-keyboardFocused & {
+    /* focus indicator is focused state */
+    &::after {
+      border-color: var(--highcontrast-textfield-focus-indicator-color, var(--mod-textfield-focus-indicator-color, var(--spectrum-textfield-focus-indicator-color)));
+      border-width: var(--mod-textfield-focus-indicator-width, var(--spectrum-textfield-focus-indicator-width));
+    }
+  }
+
+  .is-readOnly & {
+    &::after {
+      border: 0;
+    }
+  }
+
+  .spectrum-Textfield--quiet & {
+    &::after {
+      inline-size: 100%;
+      block-size: calc(100% + var(--mod-textfield-focus-indicator-gap, var(--spectrum-textfield-focus-indicator-gap)));
+      margin-inline-start: 0;
+      margin-block-start: 0;
       border-radius: 0;
+      border-color: var(--highcontrast-textfield-focus-indicator-color, var(--mod-textfield-focus-indicator-color, var(--spectrum-textfield-focus-indicator-color)));
+    }
+  }
+
+  .spectrum-Textfield--quiet.is-keyboardFocused & {
```

Comment:
`:focus-visible` has wide support now, do we still need to indicate keyboard focus using a class or can that pseudo do the work for us?

Reply:
Combine that with :focus to target non-keyboard focus: `*:focus:not(:focus-visible)`